[vansh284] iP

vansh284 opened the pull request:
# Didier

> “The computer was born to solve problems that did not exist before.” - Bill Gates ([source](https://pandorafms.com/blog/who-is-bill-gates/#:~:text=Bill%20Gates%3A%20%E2%80%9CThe%20computer%20was,didn't%20exist%20before.%E2%80%9D))

Thanks to Didier, you no longer have to remember the things you need to do.

Didier is:
- text-based
- easy to set up
- ~~EASY~~ SUPER EASY to use

This is all that you need to do to get started with Didier:
1. [download](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html) it.
2. double-click on the application.
3. add your tasks.
4. take a seat and let Didier manage your tasks 💯 

And it is completely **FREE OF COST**!

Features of Didier:
- [x] Manage your tasks
- [x] Manage deadlines
- [x] Keep track of events
- [ ] Set up recurring tasks (coming soon!)

If you're a Java Programmer, feel free to contribute! Didier is an open-source project. Here's the `main` method that can help get you started!

``` java
public class Didier {
    public static void main(String[] args) {
        Didier didier = new Didier("data/", "didier.txt");
        didier.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

vansh284 committed:
Rename bot

vansh284 committed:
Make bot echo user commands

vansh284 committed:
Make bot store and list user tasks

vansh284 committed:
Add java class files to .gitignore

vansh284 committed:
Add ability to mark tasks done/undone

vansh284 committed:
Add feature to track three task types

vansh284 committed:
Add javadocs for Task classes

vansh284 committed:
semi-automate testing of Duke

vansh284 committed:
Make bot handle basic errors

vansh284 committed:
Update testing to test for errors

vansh284 committed:
Change file name from Duke to Didier

vansh284 committed:
Use exceptions for error handling

vansh284 committed:
Add deleting tasks functionality

vansh284 committed:
Close scanner and change indentation for switch

vansh284 committed:
Use Enums for commands

vansh284 committed:
Overload Task constructors to take isDone parameter

vansh284 committed:
Add feature to save tasks locally

vansh284 committed:
Merge branch 'branch-Level-7'

vansh284 committed:
Teach chatbot to understand dates

vansh284 committed:
Merge branch 'branch-Level-8'

vansh284 committed:
Make the code more OOP

vansh284 committed:
Divide classes into packages

vansh284 committed:
Merge branch 'add-gradle-support'

vansh284 committed:
Add checkstyle plugin to gradle

vansh284 committed:
Add check for corruped file

vansh284 committed:
Move TaskList out of task package

vansh284 committed:
Extend FileCorruptedException to more cases

vansh284 committed:
Add JUnit tests for TaskList, Parser, Task

vansh284 committed:
Change shadowJar archiveBaseName to didier

vansh284 committed:
Add JavaDoc comments to most classes and methods

vansh284 committed:
Tweak code to follow coding standard

vansh284 committed:
Add feature to search for keywords in task list

vansh284 committed:
Solve merge conflicts with coding standard branch

vansh284 committed:
Resolve merge conflicts from merging branch-Level-9

vansh284 committed:
Change imports order and add more JavaDocs

vansh284 committed:
Implement basic GUI skeleton

vansh284 committed:
Add GUI using JavaFX

vansh284 committed:
Merge branch 'branch-Level-10'

redtailedfox commented on src/main/java/didier/Didier.java:
```diff
@@ -20,41 +20,43 @@ public class Didier {
+     * Returns whether the bot is currently active or not.
+     * @return The status of the bot.
+     */
+    public boolean getIsActive() {
+        return this.isActive;
+    }
```
isActive() should suffice i think

redtailedfox commented on src/main/java/didier/command/MarkCommand.java:
```diff
@@ -25,18 +26,32 @@ public class MarkCommand extends Command {
+    }
+
+    @Override
+    public String getBotOutput(TaskList taskList, Storage storage) throws DidierException {
```
Well Done on using the correct case for the method name

FerdiHS commented on src/main/java/didier/controller/DialogBox.java:
```diff
@@ -0,0 +1,68 @@
+        this.setAlignment(Pos.TOP_LEFT);
+        this.setBackground(new Background(new BackgroundFill(Color.LIGHTPINK, CornerRadii.EMPTY, Insets.EMPTY)));
+    }
+    public static DialogBox getUserDialog(String input, Image userImage) {
+        return new DialogBox(input, userImage);
+    }
```
Shouldn't there be JavaDoc comments for every public methods?

vansh284 replied:
Yes, thanks for pointing that out.

FerdiHS commented on text-ui-test/input.txt:
Shouldn't it be better to add test cases for the new feature, such as find and delete?

vansh284 committed:
Add assert statements to code

Assert statements ensure that assumptions made by the programmer are
valid.

Assert statements were added to validate the following assumptions:
* The task list of the Didier object obtained from the storage is not
  null
* The task returned from parsing the saved file string is not null
* The command returned from parsing the command string is not null

vansh284 committed:
Use the default branch

vansh284 committed:
Extract variables to make code obvious

vansh284 committed:
Refactor code to improve code quality

The following steps were taken to improve code quality:
* structure the code in the Dialog class more logically with line breaks
  and comments separating each section.
* significantly shorten the length of the Parser class's parse method by
  extracting methods and following the SLAP.
* made the happy path more prominent and included default case in some
  of the Parser class's methods.

vansh284 committed:
Make happy path more prominent in AddCommand class

vansh284 committed:
Merge pull request #1 from vansh284/branch-A-Assertions

vansh284 committed:
Merge master into branch-A-CodeQuality

vansh284 committed:
Merge pull request #2 from vansh284/branch-A-CodeQuality

Improve code quality

vansh284 committed:
Disallow adding duplicate tasks

vansh284 committed:
Fix bug with adding tasks

vansh284 committed:
Add handling error of time mismatches for the event

vansh284 committed:
Fix bug with text truncation of messages

vansh284 committed:
Add Ui.png

vansh284 committed:
Add user guide

vansh284 committed:
Update user guide

vansh284 committed:
Add support for linux machines
